docker/Dockerfile

```diff
@@ -22,26 +28,33 @@ RUN apt-get update \
     unrar-free \
     p7zip-full \
     dos2unix \
+    && npm set progress=false && npm config set depth 0 \
     && npm install -g npm@latest \
     && npm install -g pm2 \
-    && cd /enigma-bbs && npm install \
+    && cd /enigma-bbs && npm install
+
+
+# Do this after npm install to avoid cache-miss on every code change
+COPY . /enigma-bbs
+
+# Then run post source copy steps that have to happen every time
+RUN dos2unix /enigma-bbs/docker/bin/docker-entrypoint.sh \
+    && apt-get remove dos2unix -y \
+    && chmod +x /enigma-bbs/docker/bin/docker-entrypoint.sh \
+    && cp -f /enigma-bbs/docker/bin/sexyz /usr/local/bin \
+    && cd /enigma-bbs \
     && pm2 start main.js \
     && mkdir -p /enigma-bbs-pre/art \
     && mkdir /enigma-bbs-pre/mods \
     && mkdir /enigma-bbs-pre/config \
     && cp -rp art/* ../enigma-bbs-pre/art/ \
     && cp -rp mods/* ../enigma-bbs-pre/mods/ \
     && cp -rp config/* ../enigma-bbs-pre/config/ \
-    && apt-get remove build-essential python python3 libssl-dev git curl -y \
+    && apt-get remove build-essential python3 libssl-dev git curl -y \
```

Comment:
Maybe consider a multi-stage build/runtime, as this doesn't decrease the actual download size for image slices.